crates/sui-graphql-rpc/src/types/dynamic_field.rs

```diff
@@ -172,66 +160,119 @@ impl DynamicField {
         };
 
         let field_id = derive_dynamic_field_id(parent, &type_, &name.bcs.0)
-            .map_err(|e| Error::Internal(format!("Failed to derive dynamic field id: {e}")))?
-            .to_vec();
-
-        let stored: Option<StoredObject> = db
-            .execute(move |conn| {
-                conn.first(move || dsl::objects.filter(dsl::object_id.eq(field_id.clone())))
-                    .optional()
-            })
-            .await
-            .map_err(|e| Error::Internal(format!("Failed to fetch object: {e}")))?;
-
-        stored.map(Self::try_from).transpose()
+            .map_err(|e| Error::Internal(format!("Failed to derive dynamic field id: {e}")))?;
+
+        let super_ = MoveObject::query(
+            db,
+            SuiAddress::from(field_id),
+            match checkpoint_viewed_at {
+                Some(seq) => ObjectLookupKey::LatestAt(seq),
+                None => ObjectLookupKey::Latest,
+            },
+        )
+        .await?;
+
+        super_.map(Self::try_from).transpose()
     }
 
     /// Query the `db` for a `page` of dynamic fields attached to object with ID `parent`.
     pub(crate) async fn paginate(
         db: &Db,
         page: Page<object::Cursor>,
         parent: SuiAddress,
+        version: Option<u64>,
```

Comment:
nit: rename to `parent_version` and update the docs to explain what this is used for and why.